include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -23,50 +23,60 @@
-
 #include <seqan3/alphabet/gap/gapped.hpp>
-
 #include <seqan3/core/algorithm/all.hpp>
 #include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/parallel/execution.hpp>
 #include <seqan3/range/view/persist.hpp>
-
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
 
 namespace seqan3
 {
 
-/*!\brief Computes a pairwise alignment over a sequence pair or a range of sequence pairs.
+/*!\brief Computes the pairwise alignment for a pair of sequences or a range over sequence pairs.
  * \ingroup pairwise_alignment
- *
- * \tparam sequence_t         The type of sequence pairs. Either a type modelling seqan3::TupleLike
- *                            with a tuple size of exactly two or std::ranges::InputRange whose value type models the
- *                            before mentioned tuple concept.
+ * \tparam exec_policy_t      The type of the execution policy; std::is_execution_policy_v must evaluate to true.
+ * \tparam sequence_t         The type of sequence pairs (see details for more information on the type constraints).
  * \tparam alignment_config_t The type of the alignment configuration; must be a seqan3::configuration.
- * \param[in] seq             A tuple with two sequences or an input range over such tuples.
+ * \param[in] exec            The execution policy to use. Defaults to std::execution::seq.
+ * \param[in] seq             A tuple with two sequences or a range over such tuples.
  * \param[in] config          The object storing the alignment configuration.
  * \return A seqan3::alignment_range.
  *
  * \details
  *
- * This function computes the pairwise alignment of a sequence pair or a range over sequence pairs.
- * The pairs can be any class template that models the seqan3::TupleLike.
- * In case of calling the alignment with a single tuple, the value is passed as a std::view::single to the range
- * aware interface. In this case the passed class must be std::CopyConstructible and both element types of the tuple
- * must model std::ranges::ViewableRange because it is copied into the std::view::single. The same
- * is true, if the input range is a temporary non-view range.
+ * This function computes the pairwise alignment for the given sequences. During the setup phase the most efficient
+ * implementation is selected depending on the configurations stored in the given seqan3::configuration object.
+ * In addition, the alignments can be computed in parallel if std::execution::par is given as the execution policy.
+ * On default the algorithm uses the sequenced execution (std::execution::seq).
+ *
+ * ### Compute a single alignment
+ *
+ * In some cases if only a single alignment needs to be computed, the two sequences can be passed as a pair.
+ * The pairs can be any class template that models the seqan3::TupleLike concept. The tuple elements must model
+ * std::ranges::ViewableRange and std::CopyConstructible as the tuple is converted into a range using std::view::single
```

Comment:
```suggestion
 * std::ranges::ViewableRange and std::CopyConstructible. 
```

The rest is detail and it's not necessary for users to know why I would say.